wgpu/src/api/queue.rs

```diff
@@ -108,7 +108,10 @@ impl Queue {
     ///
     /// If possible, consider using [`Queue::write_buffer_with`] instead. That
     /// method avoids an intermediate copy and is often able to transfer data
-    /// more efficiently than this one.
+    /// more efficiently than this one. Currently on native platforms, for both
+    /// of these methods the staging memory will be a new allocation. This will
+    /// then be released after the next submission. To entirely avoid short-lived
+    /// allocations, you might be able to use [`StagingBelt`](crate::util::StagingBelt).
```

Comment:
this should be a new paragraph

```suggestion
    /// more efficiently than this one. 
    ///
    /// Currently on native platforms, for both of these methods the staging 
    /// memory will be a new allocation. This will then be released after the
    /// next submission. To entirely avoid short-lived allocations, you might
    /// be able to use [`StagingBelt`](crate::util::StagingBelt).
```

Reply:
> This will then be released after the next submission

This actually won't be release until a) the submission is finished b) the cpu realizes the submission is finished. Then the memory and buffer will be released. The copy to gpu memory happens inside the submission, so we cannot delete that buffer until we are sure that the gpu is finished using it.